Source/Core/GoogleEarthEnterpriseMetadata.js

```diff
@@ -0,0 +1,579 @@
+        // guess. Security through obscurity. :-(
+
+        // while we have a full uint64 (8 bytes) left to do
+        // assumes buffer is 64bit aligned (or processor doesn't care)
+        while (dp < dpend64) {
+            // rotate the key each time through by using the offets 16,0,8,16,0,8,...
+            off = (off + 8) % 24;
+            kp = off;
+
+            // run through one key length xor'ing one uint64 at a time
+            // then drop out to rotate the key for the next bit
+            while ((dp < dpend64) && (kp < kpend)) {
+                dataView.setUint32(dp, dataView.getUint32(dp, true) ^ keyView.getUint32(kp, true), true);
+                dataView.setUint32(dp + 4, dataView.getUint32(dp + 4, true) ^ keyView.getUint32(kp + 4, true), true);
+                dp += 8;
+                kp += 24;
+            }
+        }
+
+        // now the remaining 1 to 7 bytes
+        if (dp < dpend) {
+            if (kp >= kpend) {
+                // rotate the key one last time (if necessary)
+                off = (off + 8) % 24;
+                kp = off;
+            }
+
+            while (dp < dpend) {
+                dataView.setUint8(dp, dataView.getUint8(dp) ^ keyView.getUint8(kp));
+                dp++;
+                kp++;
+            }
+        }
+    };
+
+    var taskProcessor = new TaskProcessor('decodeGoogleEarthEnterprisePacket', Number.POSITIVE_INFINITY);
+
+    /**
+     * Retrieves a Google Earth Enterprise quadtree packet.
+     *
+     * @param {String} [quadKey=''] The quadkey to retrieve the packet for.
+     * @param {Number} [version=1] The cnode version to be used in the request.
+     * @param {Boolean} [throttle=true] True if the number of simultaneous requests should be limited,
+     *                  or false if the request should be initiated regardless of the number of requests
+     *                  already in progress.
+     *
+     * @private
+     */
+    GoogleEarthEnterpriseMetadata.prototype.getQuadTreePacket = function(quadKey, version, throttle) {
+        version = defaultValue(version, 1);
+        quadKey = defaultValue(quadKey, '');
+        throttle = defaultValue(throttle, true);
+        var url = getMetadataUrl(this, quadKey, version);
+        var proxy = this._proxy;
+        if (defined(proxy)) {
+            url = proxy.getURL(url);
+        }
+
+        var promise;
+        if (throttle) {
+            promise = throttleRequestByServer(url, loadArrayBuffer);
+            if (!defined(promise)) {
+                return undefined;
+            }
+        } else {
+            promise = loadArrayBuffer(url);
+        }
+
+        var tileInfo = this._tileInfo;
+        return promise
+            .then(function(metadata) {
+                var decodePromise = taskProcessor.scheduleTask({
+                    buffer : metadata,
+                    quadKey : quadKey,
+                    type : 'Metadata'
+                });
```

Comment:
Same comment for anywhere else you're doing scheduleTask and don't need to continue to use the buffer after the fact